### PR TITLE
task_DEV-3

### PR DESCRIPTION
**Pull request by TatianaGantievskaya**

### CODE REVIEW

File: NumberConverter/Converter.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace NumberConverter
+{
+  /// <summary>
+  /// This class was made for converting the numbers to different numeral systems.
+  /// </summary>
+  class Converter
+  {
+    private Int64 convertibleNumber;
+    private Int16 radix;
```
Comment (IgorKheidorov) [inline NumberConverter/Converter.cs]: no need to store radix in Int16 format

File: NumberConverter/ConvertingException.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace NumberConverter
+{
+  /// <summary>
```
Comment (IgorKheidorov) [inline NumberConverter/ConvertingException.cs]: No need to select new type of exception

File: NumberConverter/EntryPoint.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  /// </summary>
+  class EntryPoint
+  {
+    static void Main(string[] args)
+    {
+      try
+      {
+        Converter convertibleNumber = new Converter(args[0], args[1]);
+        Console.WriteLine(convertibleNumber.ConvertNumberToSelectedNumeralSystem(convertibleNumber.ChooseTheNumeralSystem()));
+      }
```
Comment (IgorKheidorov) [inline NumberConverter/EntryPoint.cs]: No need to process different types of exceptions which are processed in the same way

File: NumberConverter/EntryPoint.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  /// <summary>
+  /// The entry point of the program.
+  /// </summary>
+  class EntryPoint
+  {
+    static void Main(string[] args)
+    {
+      try
+      {
+        Converter convertibleNumber = new Converter(args[0], args[1]);
```
Comment (IgorKheidorov) [inline NumberConverter/EntryPoint.cs:14]: Very strange construction

File: NumberConverter/Converter.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    /// This method converts the number to selected numeral system.
+    /// </summary>
+    /// <param name="chosenNumeralSystem"> String, containing values of selected numeral system. </param>
+    /// <returns> It returns converted number. </returns>
+    public char[] ConvertNumberToSelectedNumeralSystem(string chosenNumeralSystem)
+    {
+      StringBuilder convertedNumber = new StringBuilder();
+      while (convertibleNumber != 0)
+      {
+        Int64 temporaryVariable = convertibleNumber / radix;
```
Comment (IgorKheidorov) [inline NumberConverter/Converter.cs]: Too complicated and unclear method to find the necessary symbol, has to be simplified

File: NumberConverter/Converter.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    {
+      convertibleNumber = Convert.ToInt64(numberToConvert);
+      radix = Convert.ToInt16(basisForConverting);
+      if (radix < 2 || radix > 20 || convertibleNumber <= 0)
+      {
+        throw new ConvertingException();
+      }
+    }
+
+    /// <summary>
```
Comment (IgorKheidorov) [inline NumberConverter/Converter.cs]: Unclear comment

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Too original decision, try to simplify

File: NumberConverter/Converter.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+using System.Text;
+
+namespace NumberConverter
+{
+  /// <summary>
+  /// This class was made for converting the numbers to different numeral systems.
+  /// </summary>
+  class Converter
+  {
+    private Int64 convertibleNumber;
```
Comment (AlexanderSoroka) [inline NumberConverter/Converter.cs]: Either use big integer either just int, int 64 has almost the same limitations as int

File: NumberConverter/Converter.cs [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    private int radix;
+
+    /// <summary>
+    /// Constructor with the parameters, which initializes class's fields.
```
Comment (AlexanderSoroka) [inline NumberConverter/Converter.cs]: "Constructor with the parameters" - it's obvious from the code

File: NumberConverter/Converter.cs [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    /// </summary>
+    /// <param name="numberToConvert"> Number, which user wants to convert. </param>
+    /// <param name="basisForConverting"> Radix. </param>
+    public Converter(string numberToConvert, string basisForConverting)
+    {
+      convertibleNumber = Convert.ToInt64(numberToConvert);
+      radix = Convert.ToInt32(basisForConverting);
+      if (radix < 2 || radix > 20 || convertibleNumber <= 0)
```
Comment (AlexanderSoroka) [inline NumberConverter/Converter.cs]: I would separate these checks

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: NumberConverter/Converter.cs [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    /// <summary>
+    /// Constructor, which initializes class's fields.
+    /// </summary>
+    /// <param name="numberToConvert"> Number, which user wants to convert. </param>
+    /// <param name="basisForConverting"> Radix. </param>
+    public Converter(string numberToConvert, string basisForConverting)
+    {
+      convertibleNumber = BigInteger.Parse(numberToConvert);
+      radix = int.Parse(basisForConverting);
+      if (radix < 2 || radix > 20)
```
Comment (AlexanderSoroka) [inline NumberConverter/Converter.cs:24]: it's better to create custom setter

File: NumberConverter/Converter.cs [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+      {
+        throw new Exception("Error, you entered incorrect convertible number.");
+      }
+    }
+
+    /// <summary>
+    /// This method converts the number to selected numeral system.
+    /// </summary>
+    /// <returns> It returns converted number. </returns>
+    public char[] ConvertNumberToSelectedNumeralSystem()
```
Comment (AlexanderSoroka) [inline NumberConverter/Converter.cs:38]: return string instead of char[]

File: NumberConverter/Converter.cs [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    {
+      StringBuilder convertedNumber = new StringBuilder();
+      while (convertibleNumber != 0)
+      {
+        BigInteger temporaryVariable = convertibleNumber / radix;
+        BigInteger calculatedVariable = convertibleNumber - radix * temporaryVariable;
+        if (calculatedVariable > 9)
+        {
+          convertedNumber.Append((char)(calculatedVariable + 55));
```
Comment (AlexanderSoroka) [inline NumberConverter/Converter.cs:47]: why 55?

**Review by AlexanderSoroka — APPROVED** [review]

**Pull request merged**